app/models/namespace_group_link.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# entity class for NamespaceGroupLink
+class NamespaceGroupLink < ApplicationRecord
+  has_logidze
+  acts_as_paranoid
+
+  before_save :set_namespace_type
```

Comment:
**nitpick (blocking):** Can this be changed to `before_validation` that way we can add validation to `namespace_type` that it only allows `Group` or `Project` types

Reply:
Sure! Updated in 7d25039d987c3189e3b66f9aa17dc306eed0acf1